### ISSUE TITLE
Accessibility of links in exception widget

### ISSUE THREAD

**Issue by kieferrm** [microsoft/vscode#111758]
Testing #110567

I tested this with Narrator on Windows.  I tried to tab to the links in the exception widget to open them, but I couldn't find a way to get there. I think I should be able to as Narrator reads the stack trace to me.

![image](https://user-images.githubusercontent.com/4674940/100904499-ee62b480-347b-11eb-80d2-bfafa602ddf0.png)

**Comment by isidorn** [comment]
I agree with this 100%. 
Assigning to December to try to add it then - since we never had this, no need to rush something in Endgame.

**Comment by jvesouza** [comment]
When opening the test execution report I find a link to go to the test source. I press the enter key on that link but nothing happens. I don't know if it's a different problem or caused by the same situation.

**Comment by isidorn** [comment]
@jvesouza that is a different problem. I am not sure what is the test exceution report, does it come from the Java extension or from the test viewlet.
fyi @connor4312

**Comment by isidorn** [comment]
Just for clarity here's the issue @jvesouza was metnionig (it is a java extension test view) https://github.com/notifications?query=is%3Adone

I have changed the exception widget in vscode that each link can be focused to, and if we know where the link leads to (we could resovle it) then `enter` and `space` open that file in the editor.

So to verify:
* Have an exception in your simple javascript program
* Make sure you can `tab` over stack trace links and that enter or space navigates you to the link (if the link points to an existing file)

**Comment by jvesouza** [comment]
I tried to test causing an exception in a java application but I was not successful.
The exception is read by orca but the tab key has no effect.
As I am not javascript knowledgeable would it be possible to give me a hint on how to cause an exception in a JS?
Perhaps this will help me to understand the problem.

**Comment by isidorn** [comment]
@jvesouza here's an example JS program that throws an exception. Just make sure to have checked "Uncaught Exceptions"

```
throw new Error('hello there')
```

What happens to me. Once an exception is thrown, focus is on the exception widget and I can press `tab`, first focus on the `Esc` button and following `tab` to navgiate over the links.

**Comment by jvesouza** [comment]
I have the following java code:


> class Test {
>    public static void main(String[] args) {
>    args[10] = null;
>    }
> }


When I debug this code I receive the following error
> Exception has occurred: java.lang.ArrayIndexOutOfBoundsException
> "java.lang.ArrayIndexOutOfBoundsException: Index 10 out of bounds for length 0"

But when I press the tab key no links are shown.

**Comment by isidorn** [comment]
Maybe that exception does not show a stack trace. I do not have java setup so I can not really tell.
So either you can try with JavaScript or I can try to use a Docer Container that has Java setup

**Comment by jvesouza** [comment]
I can confirm that using JavaScript the links are displayed as described. However the same does not happen when using a Java application.
Is there anything I can do to help identify this likely problem?
Send a print screen, record a video, run code-insiders with some kind of debug  or something worthwhile.

**Comment by isidorn** [comment]
@jvesouze you can send a print screen, or maybe @jdneo knows more. If the stack trace information is part of java exception stack trace

**Comment by jdneo** [comment]
Yes a screenshot will be helpful!

**Comment by jvesouza** [comment]
I'm trying to send two screenshots. I'm sorry if they are not ok because as I am blind, I made the capture blindly!
Let me know if they are not ok, I can make a new capture with the help of someone who sees.

The first image shows the result when I used a JS application and the second shows the result when I used a java application.
![test js](https://user-images.githubusercontent.com/2037394/105331303-0fa96c80-5bb2-11eb-89de-83cafd16c050.png)
![Test java](https://user-images.githubusercontent.com/2037394/105331396-2cde3b00-5bb2-11eb-9ab1-d025642f9c39.png)

**Comment by isidorn** [comment]
@jvesouza thanks for the screen shots, they show the issue.

The javascript throws a stack trace and thus you can tab into it.
In the Java example there is not stack trace, and thus you can not tab.
This is controlled by the java extension, so @jdneo or @testforstephen could comment why in this example the exception does not contain a stack trace. To send a stack trace with the exception an adapter has to fill this [`stackTrace`](https://github.com/microsoft/vscode/blob/c71edb7d88335f440c7b1fa5619d333c34fd7854/src/vs/workbench/contrib/debug/common/debugProtocol.d.ts#L2168)

**Comment by testforstephen** [comment]
@isidorn @jvesouza thanks, i have created a feature request at https://github.com/microsoft/vscode-java-debug/issues/940 to print stacktraces info for exception breakpoint.

**Comment by isidorn** [comment]
@testforstephen awesome, thanks

**Comment by rzhao271** [comment]
On both Codespaces and Windows I'm unable to tab into the links with Narrator.
The focus order is the red Exception view, followed by the close button for that view, followed by the Problems tab in the panel that I have on the bottom.
